config/aso/kustomization.yaml

```diff
@@ -22,18 +22,16 @@ patches:
       version: v1
       kind: Deployment
       name: azureserviceoperator-controller-manager
-  - patch: |- # remove permissions to manage CRDs
-      $patch: delete
-      apiVersion: rbac.authorization.k8s.io/v1
-      kind: ClusterRole
-      metadata:
-        name: azureserviceoperator-crd-manager-role
-  - patch: |-
-      $patch: delete
-      apiVersion: rbac.authorization.k8s.io/v1
-      kind: ClusterRoleBinding
-      metadata:
-        name: azureserviceoperator-crd-manager-rolebinding
+  - patch: |- # Users can specify additional ASO CRDs. CRDs should be appended with ';'
+      - op: replace
+        path: /spec/template/spec/containers/0/args/4
+        value: --crd-pattern=${ADDITIONAL_ASO_CRDS:-}
```

Comment:
Can we add this to the existing patch above so we don't have to redefine `target`?

Reply:
It would definitely read better if it looked like this:

```yaml
  - patch: |- # CAPZ will manage ASO's CRDs
    target:
      group: apps
      version: v1
      kind: Deployment
      name: azureserviceoperator-controller-manager
      - op: test
        path: /spec/template/spec/containers/0/args/4
        value: --crd-pattern=
      - op: replace
        path: /spec/template/spec/containers/0/args/4
        value: --crd-pattern=${ADDITIONAL_ASO_CRDS:-}
```

In other words, putting the `op`s after the `target` definition.

Reply:
> putting the ops after the target definition.

Refactoring it that way results in _"Schema validation: Incompatible types. Required: array. Actual: null."_

Reply:
I think Jack's syntax might be off a bit.

Reply:
```
  - patch: |-
      - op: test
        path: /spec/template/spec/containers/0/args/4
        value: --crd-pattern=
      - op: replace
        path: /spec/template/spec/containers/0/args/4
        value: --crd-pattern=${ADDITIONAL_ASO_CRDS:-}
    target:
      group: apps
      version: v1
      kind: Deployment
      name: azureserviceoperator-controller-manager

```

Reply:
^Put this already, performing a sanity check.